src/metabase/upload.clj

```diff
@@ -797,33 +798,36 @@
         (ex-info (tru "You don''t have permissions to do that.")
                  {:status-code 403}))))
 
-(defn- check-can-append
+(defn- check-can-update
   "Throws an error if the user cannot upload to the given database and schema."
   [db table]
-  (when-let [error (can-append-error db table)]
+  (when-let [error (can-update-error db table)]
     (throw error)))
 
 (defn can-upload-to-table?
   "Returns true if the user can upload to the given database and table, and false otherwise."
   [db table]
-  (nil? (can-append-error db table)))
+  (nil? (can-update-error db table)))
 
 ;;; +--------------------------------------------------
-;;; |  public interface for appending to uploaded table
+;;; |  public interface for updating an uploaded table
 ;;; +--------------------------------------------------
 
-(mu/defn append-csv!
-  "Main entry point for appending to uploaded tables with a CSV file.
+(mu/defn update-csv!
+  "Main entry point for updating an uploaded table with a CSV file.
   This will create an auto-incrementing primary key (auto-pk) column in the table for drivers that supported uploads
-  before auto-pk columns were introduced by metabase#36249."
-  [{:keys [^File file table-id]}
+  before auto-pk columns were introduced by metabase#36249, if it does not already exist."
+  ;; TODO: update callers to all pass an explicit action, and make it required
+  [{:keys [^File file table-id action] :or {action ::append}}
    :- [:map
        [:table-id ms/PositiveInt]
-       [:file (ms/InstanceOfClass File)]]]
+       [:file (ms/InstanceOfClass File)]
+       [:action keyword?]]]
```

Comment:
would be good to lock this down to a set of values sooner rather than later

Reply:
No sure why I was so lax